app/controllers/application_controller.rb

```diff
@@ -25,10 +25,14 @@ def current_user
 
   def authenticate
     if current_user
+      assign_sentry_contexts
       add_token_to_connection
       set_has_multiple_providers
 
-      set_user_session if current_user['user_id'].blank?
+      if current_user['user_id'].blank?
+        set_user_session
+        Raven.user_context(id: current_user['user_id'])
```

Comment:
Hmm, can we just run `assign_sentry_contexts` on line 35 after the `if`, so it'll always get set when we have data?

Reply:
I wanted to have the context set ASAP ... with the idea being that if `add_token_to_connection` or `set_has_multiple_providers` blow up, we still have _some_ context.